BUILD

```diff
@@ -18,64 +19,37 @@ touch $@
     stamp = True,
 )
 
-alias(
-    name = "android_pom",
-    actual = "//library/kotlin/src/io/envoyproxy/envoymobile:android_aar_pom",
-)
-
 alias(
     name = "android_aar",
-    actual = "//library/kotlin/src/io/envoyproxy/envoymobile:android_aar",
-)
-
-alias(
-    name = "android_javadocs",
-    actual = "//library:javadocs",
-)
-
-alias(
-    name = "android_sources",
-    actual = "//library:sources_jar",
+    actual = "@envoy_mobile//library/kotlin/src/io/envoyproxy/envoymobile:android_aar_only_aar",
+    visibility = ["//visibility:public"],
 )
 
 genrule(
-    name = "android_dist",
+    name = "android_zip",
     srcs = [
-        "android_aar",
+        "@envoy_mobile//library/kotlin/src/io/envoyproxy/envoymobile:android_aar",
     ],
-    outs = ["stub_android_dist_output"],
-    cmd = """
-cp $(location :android_aar) dist/envoy.aar
-chmod 755 dist/envoy.aar
-touch $@
-""",
+    outs = ["envoy_mobile.zip"],
+    cmd = "$(location @envoy_mobile//bazel:zipper) fc $@ $(SRCS)",
     stamp = True,
+    tools = ["@envoy_mobile//bazel:zipper"],
+    visibility = ["//visibility:public"],
 )
 
 genrule(
-    name = "android_deploy",
+    name = "android_dist",
     srcs = [
-        "android_aar",
-        "android_pom",
-        "android_javadocs",
-        "android_sources",
+        "@envoy_mobile//library/kotlin/src/io/envoyproxy/envoymobile:android_aar",
     ],
-    outs = ["stub_android_deploy_output"],
+    outs = ["output_in_dist_directory"],
     cmd = """
-cp $(location :android_aar) dist/envoy.aar
-cp $(location :android_pom) dist/envoy-pom.xml
-cp $(location :android_javadocs) dist/envoy-javadoc.jar
-cp $(location :android_sources) dist/envoy-sources.jar
-chmod 755 dist/envoy.aar
-chmod 755 dist/envoy-pom.xml
-chmod 755 dist/envoy-javadoc.jar
-chmod 755 dist/envoy-sources.jar
-orig_dir=$$PWD
-pushd dist
-zip -r envoy_aar_sources.zip envoy.aar envoy-pom.xml envoy-javadoc.jar envoy-sources.jar > /dev/null
-popd
-touch $@
-""",
+    for artifact in $(SRCS); do
+        chmod 755 $$artifact
+        cp $$artifact dist/
+    done
+    touch $@
+    """,
     stamp = True,
```

Comment:
I thought you were going to toss this one entirely?

Reply:
I...need to do more work to get that